pyteal/ast/asset.py

```diff
@@ -51,6 +53,39 @@ def frozen(cls, account: Expr, asset: Expr) -> MaybeValue:
 AssetHolding.__module__ = "pyteal"
 
 
+class AssetHoldingObject:
+    """Represents information about an account's holding of an asset"""
+
+    def __init__(self, asset: Expr, account: Expr) -> None:
+        """Create a new AssetParamObject for the given asset.
+
+        Args:
+            asset: An identifier for the asset. It must be an index into Txn.ForeignAssets that
+                corresponds to the asset to check, or since v4, an asset ID that appears in
+                Txn.ForeignAssets. In either case, it must evaluate to uint64.
+            account: An identifier for the account. It must be an index into Txn.Accounts that
+                corresponds to the account to check (in which case it must evaluate to uint64), or
+                since v4, an account address that appears in Txn.Accounts or is Txn.Sender (in which
+                case it must evaluate to bytes).
+        """
+        self._asset: Final = asset
+        self._account: Final = account
```

Comment:
if the previous assumption on type check is correct, I think this place also need some type check?